src/transaction.ts

```diff
@@ -202,18 +204,193 @@ export const sendTransaction = async function (
+    const newAtc = new AtomicTransactionComposer()
+    newAtc.addTransaction({ txn: txnToSend, signer: getSenderTransactionSigner(from) })
+    const packed = await packResources(algod, newAtc)
+    txnToSend = packed.buildGroup()[0].txn
+  }
+
+  const signedTransaction = await signTransaction(txnToSend, from)
 
   await algod.sendRawTransaction(signedTransaction).do()
 
-  Config.getLogger(suppressLog).info(`Sent transaction ID ${transaction.txID()} ${transaction.type} from ${getSenderAddress(from)}`)
+  Config.getLogger(suppressLog).info(`Sent transaction ID ${txnToSend.txID()} ${txnToSend.type} from ${getSenderAddress(from)}`)
 
   let confirmation: modelsv2.PendingTransactionResponse | undefined = undefined
   if (!skipWaiting) {
-    confirmation = await waitForConfirmation(transaction.txID(), maxRoundsToWaitForConfirmation ?? 5, algod)
+    confirmation = await waitForConfirmation(txnToSend.txID(), maxRoundsToWaitForConfirmation ?? 5, algod)
+  }
+
+  return { transaction: txnToSend, confirmation }
+}
+
+/**
+ * Get all of the unamed resources used by the group in the given ATC
+ *
+ * @param algod The algod client to use for the simulation
+ * @param atc The ATC containing the txn group
+ * @returns The unnamed resources accessed by the group and by each transaction in the group
+ */
+export async function getUnnamedResourcesAccessed(algod: algosdk.Algodv2, atc: algosdk.AtomicTransactionComposer) {
+  const simReq = new algosdk.modelsv2.SimulateRequest({
+    txnGroups: [],
+    allowUnnamedResources: true,
+  })
+
+  const result = await atc.simulate(algod, simReq)
+
+  const groupResponse = result.simulateResponse.txnGroups[0]
+
+  if (groupResponse.failureMessage) {
+    throw Error(`Error during resource packing simulation in transaction ${groupResponse.failedAt}: ${groupResponse.failureMessage}`)
+  }
+
+  return {
+    group: groupResponse.unnamedResourcesAccessed,
+    txns: groupResponse.txnResults.map(
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (t: any) => t.unnamedResourcesAccessed,
+    ) as algosdk.modelsv2.SimulateUnnamedResourcesAccessed[],
+  }
+}
+
+/**
+ * Take an existing ATC and return a new ATC with packed resources
+ *
+ * @param algod The algod client to use for the simulation
+ * @param atc The ATC containing the txn group
+ * @returns A new ATC with the resources packed into the transactions
+ *
+ * @privateRemarks
+ *
+ * This entire function will eventually be implemented in simulate upstream in algod. The simulate endpoint will return
+ * an array of refference arrays for each transaction, so this eventually will eventually just call simulate and set the
+ * reference arraysm in the transactions to the reference arrays returned by simulate.
+ *
+ * See https://github.com/algorand/go-algorand/pull/5684
+ *
+ */
+export async function packResources(algod: algosdk.Algodv2, atc: algosdk.AtomicTransactionComposer) {
```

Comment:
I wonder if `packResources` is descriptive enough, since it's exported, it'll be available in algokit.packResources.

I think a better name would be something like `packAppCallResources`